### ISSUE TITLE
LOSS,SAM

### ISSUE THREAD

**Issue by SsiXia** [qffang/UnGUN#3]
How are the coefficients of generator loss and discriminator loss adjusted in the code, as well as their alternate training situation adjusted? I changed to a new data set, the loss has been fluctuating, and the downward trend is not obvious. The evaluation index SAM in the final test is very big.

**Comment by qffang** [comment]
It may be because the spectral response function has too many 0 values, resulting in a lack of sufficient constraints on the loss, you can replace the spectral response function and try again.

**Comment by SsiXia** [comment]
Hello, besides the influence of spectral response function, is there any other reason? What I want to know most is the problem of loss coefficient, loss_pre will rise first and then fall, and finally fall to stability is not higher than the first loss. Have you ever had such a situation before? I have tried to adjust the loss coefficient of generator and the alternating iterations of generator discriminator, but it still happens.

**Comment by qffang** [comment]
In previous experiments, if pretrained weights were used, the loss would rise first and then decrease. As for the weight coefficient, it is generally to ensure that the magnitude of each losses is similar.

**Comment by SsiXia** [comment]
> In previous experiments, if pretrained weights were used, the loss would rise first and then decrease. As for the weight coefficient, it is generally to ensure that the magnitude of each losses is similar.

I did not use the pre-training weight, the loss still appears like this, and it seems that the initial effect is better every time, and the more training after the worse, I have been troubled for a long time.

**Comment by qffang** [comment]
I haven't encountered this problem before, have you tried using other datasets？发自我的 iPhone在 2023年11月21日，15:36，juzisi ***@***.***> 写道：﻿

In previous experiments, if pretrained weights were used, the loss would rise first and then decrease. As for the weight coefficient, it is generally to ensure that the magnitude of each losses is similar.

I did not use the pre-training weight, the loss still appears like this, and it seems that the initial effect is better every time, and the more training after the worse, I have been troubled for a long time.

**Comment by SsiXia** [comment]
> I haven't encountered this problem before, have you tried using other datasets？发自我的 iPhone在 2023年11月21日，15:36，juzisi ***@***.***> 写道：﻿ In previous experiments, if pretrained weights were used, the loss would rise first and then decrease. As for the weight coefficient, it is generally to ensure that the magnitude of each losses is similar. I did not use the pre-training weight, the loss still appears like this, and it seems that the initial effect is better every time, and the more training after the worse, I have been troubled for a long time. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

I haven't tried any other data sets, and I keep using the same public data set. Is it related to the data set？

**Comment by qffang** [comment]
I'm not sure why your experiments aren't good enough, so you can only do more experiments yourself to find the problem.

**Comment by SsiXia** [comment]
> I'm not sure why your experiments aren't good enough, so you can only do more experiments yourself to find the problem.

All right. Thank you.
